rest/api_test.go

```diff
@@ -5620,45 +5620,404 @@ func TestUptimeStat(t *testing.T) {
 	assert.True(t, uptime1 < uptime2)
 }
 
-func TestUserXattrSyncFn(t *testing.T) {
+func TestUserXattrAutoImport(t *testing.T) {
 	if base.UnitTestUrlIsWalrus() {
-		t.Skip("Couchbase buckets only")
+		t.Skip("This test only works against Couchbase Server")
+	}
+
+	if !base.TestUseXattrs() {
+		t.Skip("This test only works with XATTRS enabled")
 	}
 
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+
+	docKey := t.Name()
+	xattrKey := "myXattr"
+	channelName := "testChan"
+
+	// Sync function to set channel access to whatever xattr is
 	rt := NewRestTester(t, &RestTesterConfig{
-		SyncFn: `function(doc, oldDoc, meta){if(meta.xattrs.myxattr !== undefined){channel(meta.xattrs.myxattr);}}`,
 		DatabaseConfig: &DbConfig{
-			UserXattrKey: "myxattr",
+			AutoImport:   true,
+			UserXattrKey: xattrKey,
 		},
+		SyncFn: `
+			function (doc, oldDoc, meta){
+				if (meta.xattrs.myXattr !== undefined){
+					channel(meta.xattrs.myXattr);
+				}
+			}`,
 	})
+
 	defer rt.Close()
 
 	gocbBucket, ok := base.AsGoCBBucket(rt.Bucket())
 	if !ok {
 		t.Skip("Test requires Couchbase Bucket")
 	}
 
-	var doc struct {
-		SyncData db.SyncData `json:"_sync"`
+	// Add doc
+	resp := rt.SendAdminRequest("PUT", "/db/"+docKey, "{}")
+	assertStatus(t, resp, http.StatusCreated)
+
+	// Add xattr to doc
+	_, err := base.WriteXattr(gocbBucket, docKey, xattrKey, channelName)
+	assert.NoError(t, err)
+
+	// Wait for doc to be imported
+	err = rt.WaitForCondition(func() bool {
+		return rt.GetDatabase().DbStats.SharedBucketImport().ImportCount.Value() == 1
+	})
+	assert.NoError(t, err)
+
+	// Ensure sync function has ran twice (once for PUT and once for xattr addition)
+	assert.Equal(t, int64(2), rt.GetDatabase().DbStats.CBLReplicationPush().SyncFunctionCount.Value())
+
+	// Get Xattr and ensure channel value set correctly
+	var syncData db.SyncData
+	_, err = gocbBucket.GetXattr(docKey, base.SyncXattrName, &syncData)
+	assert.NoError(t, err)
+
+	assert.Equal(t, []string{channelName}, syncData.Channels.KeySet())
+
+	// Update xattr again but same value and ensure it isn't imported again (crc32 hash should match)
+	_, err = base.WriteXattr(gocbBucket, docKey, xattrKey, channelName)
+	assert.NoError(t, err)
+
+	err = rt.WaitForCondition(func() bool {
+		return rt.GetDatabase().DbStats.Database().Crc32MatchCount.Value() == 1
```

Comment:
In other tests we look for changes in stats, as opposed to absolute values (i.e. check stat before+after desired operation).   I think this was previously due to expvars being process scoped and not necessarily being reset between tests.  Is this no longer necessary with the 2.8.0 changes to stat handling?

Reply:
The stat handling changes do mean that the stats are reset for each database context but I think this was the case before too. I believe that we only did before / after if we weren't really able to be sure about what the count would be. In this case we know that the crc match count should only be 1 as we're doing each operation sequentially and can be sure what happens when.